index.html

```diff
@@ -3012,25 +3018,35 @@ <h3>The <dfn>PropertyReadHandler</dfn> callback</h3>
 			see https://github.com/w3c/wot-scripting-api/issues/407
 		  -->
           <li>
-            Let |handler:function| be `null.`
+            Let |data| be the second argument.
           </li>
           <li>
-            If there is a user provided {{PropertyReadHandler}} in {{ExposedThing/[[readHandlers]]}}
-            <a>internal slot</a> for |interaction|, let |handler| be that.
-          </li>
-          <li>
-            If |handler| is `null`, abort these steps.
-          </li>
-          <li>
-            Let |promise| be the result of invoking |handler| with `null`.
-          </li>
-          <li>
-             If |promise| rejects, abort these steps.
+            If |data| is `undefined`, run the following sub-steps:
+            <ol>
+              <li>
+                Let |handler:function| be `null.`
+              </li>
+              <li>
+                If there is a user provided {{PropertyReadHandler}} in {{ExposedThing/[[readHandlers]]}}
+                <a>internal slot</a> for |interaction|, let |handler| be that.
+              </li>
+              <li>
+                If |handler| is `null`, abort these steps.
+              </li>
+              <li>
+                Let |promise| be the result of invoking |handler| with `null`.
+              </li>
+              <li>
+                 If |promise| rejects, abort these steps.
+              </li>
+              <li>
+                Otherwise let |data| be the value |promise| resolves with.
```

Comment:
Not sure, but for me it would read better like this 

Otherwise let |data| be the value **the** |promise| resolves with.

Reply:
|promise| is a named variable declared before, so it doesn't need "the"

Reply:
Maybe we can reformulate the whole sentence, suggestions welcome.

Reply:
It's fine by me... it just reads somehow _strange_

Reply:
It's been changed.